wfe/wfe.go

```diff
@@ -1882,6 +1891,18 @@ func (wfe *WebFrontEndImpl) FinalizeOrder(
 		return
 	}
 
+	// Ensure the signature on the CSR is good
+	if err := parsedCSR.CheckSignature(); err != nil {
+		wfe.sendError(acme.BadCSRProblem("Bad signature on CSR: "+err.Error()), response)
+		return
+	}
+
+	// Ensure a supported CSR algorithm is used
+	if good := goodSignatureAlgorithms[parsedCSR.SignatureAlgorithm]; !good {
+		wfe.sendError(acme.BadCSRProblem("CSR signature algorithm not supported: "+parsedCSR.SignatureAlgorithm.String()), response)
```

Comment:
Same here, and we usually reserve "message: other message" for cases where the bit after the colon is an embedded error:
```suggestion
		wfe.sendError(acme.BadCSRProblem(fmt.Sprintf("CSR signature algorithm %s is not supported", parsedCSR.SignatureAlgorithm.String())), response)
```